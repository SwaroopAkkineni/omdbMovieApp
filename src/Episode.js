import React, {Component} from 'react';
import {Dimensions, FlatList, Platform, Text, TouchableOpacity, View} from 'react-native';
import styles from '../StyleSheet';

export default class Episode extends React.Component {
  state = {
    episode: '',
  };

  componentDidMount() {
    this.getEpisode();
  }


  getEpisode = () => {
    const { navigation } = this.props;
    const episode = navigation.getParam('episode');
    const showName = navigation.getParam('showName');
    const season = navigation.getParam('season');
    let fetchString = 'http://www.omdbapi.com/?t=';

    showName.split(' ').forEach(function(str) {
      fetchString += str + '+'
    });
    fetchString += '&season=' + season + '&episode=' + episode +'&apikey=92e3aa84'
    console.log('fetchString: ', fetchString);
    fetch(fetchString)
      .then((response) => response.json())
      .then((responseJson) => {
        console.log('responseJson: ', responseJson);
        this.setState({
          episode: responseJson,
        });
      })
      .catch((error) => {
        console.error(error);
      });
  }


  render() {
    const {episode} = this.state;
    return (
      <View style={styles.container}>
        <Text style={[styles.Title, styles.FontColor]}>{episode.Title}</Text>
        <Text style={[styles.EpisodeInformation, styles.FontColor]}>
          <Text style={{fontWeight: 'bold'}}>Directed by: </Text>
          <Text>{episode.Director}</Text>
        </Text>
        <Text style={[styles.EpisodeInformation, styles.FontColor]}>
          <Text style={{fontWeight: 'bold'}}>Written by: </Text>
          <Text>{episode.Writer}</Text>
        </Text>
        <Text style={[styles.EpisodeInformation, styles.FontColor]}>
          <Text style={{fontWeight: 'bold'}}>Rated: </Text>
          <Text>{episode.Rated}</Text>
        </Text>
        <Text style={[styles.EpisodeInformation, styles.FontColor]}>
          <Text style={{fontWeight: 'bold'}}>Plot: </Text>
          <Text>{episode.Plot}</Text>
        </Text>
      </View>
    );
  }
}
