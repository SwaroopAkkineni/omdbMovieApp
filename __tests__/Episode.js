import React from 'react';
import 'react-native';
import Episode from '../src/Episode';

import renderer from 'react-test-renderer';
const navigationMock = { state: { params: { episode: "TestEpisode", season: "TestSeason", showName: "TestShowName" } } };

const mockResponse = (status, statusText, response) => {
        return new Response(response, {
            status: status,
            statusText: statusText,
            headers: {
                'Content-type': 'application/json'
            }
        });
    };

test('renders correctly', () => {
  fetch = jest.fn(() => new Promise(resolve => resolve()));
  const tree = renderer.create(<Episode navigation={navigationMock}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
