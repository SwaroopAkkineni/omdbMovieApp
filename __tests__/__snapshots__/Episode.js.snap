// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#89D2DC",
      "flex": 1,
      "paddingLeft": 75,
      "paddingRight": 75,
      "textAlign": "left",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 40.019999999999996,
          "fontWeight": "bold",
          "marginBottom": 37.5,
          "marginTop": 37.5,
        },
        Object {
          "color": "#101D42",
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "marginTop": 37.5,
        },
        Object {
          "color": "#101D42",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Directed by: 
    </Text>
    <Text />
  </Text>
  <Text
    style={
      Array [
        Object {
          "marginTop": 37.5,
        },
        Object {
          "color": "#101D42",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Written by: 
    </Text>
    <Text />
  </Text>
  <Text
    style={
      Array [
        Object {
          "marginTop": 37.5,
        },
        Object {
          "color": "#101D42",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Rated: 
    </Text>
    <Text />
  </Text>
  <Text
    style={
      Array [
        Object {
          "marginTop": 37.5,
        },
        Object {
          "color": "#101D42",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Plot: 
    </Text>
    <Text />
  </Text>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "height": 600,
        "marginTop": 37.5,
        "width": 600,
      }
    }
  />
</View>
`;
